src/ui/components/PasswordModule/PasswordModule.tsx

```diff
@@ -44,16 +54,43 @@ const PasswordModule = forwardRef<PasswordModuleRef, PasswordModuleProps>(
       setHintValue("");
     };
 
+    const handleClearExisting = () => {
+      setAlertExistingIsOpen(false);
+      handleClearState();
+    };
+
     useImperativeHandle(ref, () => ({
       clearState: handleClearState,
     }));
 
     const handleContinue = async (skipped: boolean) => {
-      if (!skipped) {
+      if (skipped) {
+        await Agent.agent.basicStorage.createOrUpdateBasicRecord(
+          new BasicRecord({
+            id: MiscRecordId.APP_PASSWORD_SKIPPED,
+            content: { value: skipped },
+          })
+        );
+      } else {
+        if (authentication.passwordIsSet) {
+          const currentPassword = await SecureStorage.get(
```

Comment:
One final thing: paranoid me would still catch the case where the value doesn't exist in the secure storage.

It's possible something went wrong while updating the `passwordIsSet` value in Redux - so if we catch the correct error we can recover here.

```
const currentPassword = await SecureStorage.get(
  KeyStoreKeys.APP_OP_PASSWORD
).catch((e) => {
  if (e instanceof Error && e.message === `${SecureStorage.KEY_NOT_FOUND} ${KeyStoreKeys.APP_OP_PASSWORD}`) {
    return undefined;
  }
  throw e;
});
if (currentPassword !== undefined && currentPassword === createPasswordValue) {
  setAlertExistingIsOpen(true);
  return;
}
```
^ haven't ran this but should be close enough